pkg/controller/common/featuregates.go

```diff
@@ -76,6 +77,8 @@ func IsBootImageControllerRequired(ctx *ControllerContext) bool {
 		return fg.Enabled(features.FeatureGateManagedBootImagesAWS)
 	case configv1.GCPPlatformType:
 		return fg.Enabled(features.FeatureGateManagedBootImages)
+	case configv1.VSpherePlatformType:
+		return fg.Enabled(features.FeatureGateManagedBootImages)
```

Comment:
need to change this to: `features.FeatureGateManagedBootImagesvSphere`
https://github.com/RishabhSaini/api/blob/463b93196fdb153bf0c2d5fd702990931a83d299/features/features.go#L412